server/chat/backend/agent/tools/vcs_rca_utils.py

```
@@ -0,0 +1,185 @@
+
+    Returns:
+        (repo_path_or_None, source_description)
+        For GitHub: repo_path is 'owner/repo'
+        For GitLab: repo_path is 'namespace/project'
+    """
+    if explicit_repo:
+        if provider == "github":
+            parts = explicit_repo.split('/')
+            if len(parts) == 2:
+                return explicit_repo, "explicit parameter"
+            logger.warning(f"Invalid repo format: {explicit_repo}")
```

Ruff (0.15.12)

[warning] Logging statement uses f-string

(G004)